compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -162,6 +163,40 @@ struct QueryNormalizer<'cx, 'tcx> {
     universes: Vec<Option<ty::UniverseIndex>>,
 }
 
+impl<'cx, 'tcx> QueryNormalizer<'cx, 'tcx> {
+    fn fold_reveal(
+        &mut self,
+        ty: Ty<'tcx>,
+        def_id: DefId,
+        substs: SubstsRef<'tcx>,
+    ) -> Result<Ty<'tcx>, <Self as FallibleTypeFolder<'tcx>>::Error> {
+        let substs = substs.try_fold_with(self)?;
+        let recursion_limit = self.tcx().recursion_limit();
+        if !recursion_limit.value_within_limit(self.anon_depth) {
```

Comment:
Hmm, I'm pretty sure the "anon" in `anon_depth` should be "opaque" now. Which makes the use here kind of a misnomer.